File: app/src/main/java/com/blues/module_register/RegisterActivity.java
package com.blues.module_register;

import android.os.Bundle;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentTransaction;
import android.view.View;
import android.widget.FrameLayout;
import android.widget.TextView;

import com.blues.R;
import com.blues.framework.base.BaseActivity;

import java.util.ArrayList;
import java.util.List;

/**
 * User : Blues
 * Date : 2019/5/8
 * Time : 10:21
 */

public class RegisterActivity extends BaseActivity implements View.OnClickListener {

    private List<Fragment> fragments = new ArrayList<>();
    private Fragment mCurrentFragment;

    private TextView tvAccountRegister, tvPhoneRegister;
    private FrameLayout flContainer;

    @Override
    public int setLayoutResourceId() {
        return R.layout.activity_register;
    }

    @Override
    public void initView(Bundle savedInstanceState) {

        tvAccountRegister = findViewById(R.id.tv_account_register);
        tvPhoneRegister = findViewById(R.id.tv_phone_register);
        flContainer = findViewById(R.id.fl_container);

        fragments.add(new AccountRegisterFragment());
        fragments.add(new PhoneRegisterFragment());


    }

    @Override
    public void setListener() {
        tvAccountRegister.setOnClickListener(this);
        tvPhoneRegister.setOnClickListener(this);
    }

    private void showFragment(Fragment from, Fragment to) {
        if (to == null) return;
        FragmentTransaction transaction = getSupportFragmentManager().beginTransaction();
        boolean isAdded = to.isAdded();
        if (!isAdded) {
            if (from != null) {
                transaction.hide(from)
                        .add(R.id.fl_container, to, null)
                        .show(to)
                        .commitAllowingStateLoss();
            } else {
                transaction.add(R.id.fl_container, to, null)
                        .show(to)
                        .commitAllowingStateLoss();
            }
        } else {
            if (from != null) {
                transaction.hide(from)
                        .show(to)
                        .commitAllowingStateLoss();
            } else {
                transaction.show(to)
                        .commitAllowingStateLoss();
            }
        }

        mCurrentFragment = to;
    }

    @Override
    public void onClick(View v) {
        switch (v.getId()) {
            case R.id.tv_account_register:
                //TODO 跳转账号注册fragment
                tvAccountRegister.setVisibility(View.GONE);
                tvPhoneRegister.setVisibility(View.GONE);
                flContainer.setVisibility(View.VISIBLE);
                showFragment(mCurrentFragment, fragments.get(0));
                break;
            case R.id.tv_phone_register:
                //TODO 跳转手机注册fragment
                tvAccountRegister.setVisibility(View.GONE);
                tvPhoneRegister.setVisibility(View.GONE);
                flContainer.setVisibility(View.VISIBLE);
                showFragment(mCurrentFragment, fragments.get(1));
                break;
        }
    }
}

File: app/src/main/java/com/blues/module_register/view/RegisterActivity.kt
package com.blues.module_register.view

import com.blues.framework.base.BaseKoinActivity
import android.widget.TextView
import android.widget.FrameLayout
import com.blues.R
import android.os.Bundle
import android.view.View
import androidx.fragment.app.Fragment
import com.blues.module_register.view.AccountRegisterFragment
import com.blues.module_register.view.PhoneRegisterFragment
import org.w3c.dom.Text
import java.util.ArrayList

/**
 * User : Blues
 * Date : 2019/5/8
 * Time : 10:21
 */
class RegisterActivity : BaseKoinActivity(), View.OnClickListener {

    private val fragments: List<Fragment> = listOf(AccountRegisterFragment(), PhoneRegisterFragment())
    private var mCurrentFragment: Fragment? = null
    private lateinit var tvAccountRegister: TextView
    private lateinit var tvPhoneRegister: TextView
    private lateinit var flContainer: FrameLayout

    private fun showFragment(from: Fragment?, to: Fragment?) {
        if (to == null) return
        val transaction = supportFragmentManager.beginTransaction()
        val isAdded = to.isAdded
        if (!isAdded) {
            if (from != null) {
                transaction.hide(from)
                    .add(R.id.fl_container, to, null)
                    .show(to)
                    .commitAllowingStateLoss()
            } else {
                transaction.add(R.id.fl_container, to, null)
                    .show(to)
                    .commitAllowingStateLoss()
            }
        } else {
            if (from != null) {
                transaction.hide(from)
                    .show(to)
                    .commitAllowingStateLoss()
            } else {
                transaction.show(to)
                    .commitAllowingStateLoss()
            }
        }
        mCurrentFragment = to
    }

    override fun onClick(v: View) {
        when (v.id) {
            R.id.tv_account_register -> { //TODO 跳转账号注册fragment
                tvAccountRegister.visibility = View.GONE
                tvPhoneRegister.visibility = View.GONE
                flContainer.visibility = View.VISIBLE
                showFragment(mCurrentFragment, fragments[0])
            }
            R.id.tv_phone_register -> { //TODO 跳转手机注册fragment
                tvAccountRegister.visibility = View.GONE
                tvPhoneRegister.visibility = View.GONE
                flContainer.visibility = View.VISIBLE
                showFragment(mCurrentFragment, fragments[1])
            }
        }
    }

    override fun getLayoutId(): Int {
        return R.layout.activity_register
    }

    override fun initData(savedInstanceState: Bundle?) {
        tvAccountRegister = findViewById<TextView>(R.id.tv_account_register).apply {
            setOnClickListener(this@RegisterActivity)
        }
        tvPhoneRegister = findViewById<TextView>(R.id.tv_phone_register).apply {
            setOnClickListener(this@RegisterActivity)
        }
        flContainer = findViewById(R.id.fl_container)
    }
}